src/plugin/withAndroidSourcemaps/withAndroidSourcemaps.ts

```diff
@@ -7,22 +7,52 @@
 import type { ConfigPlugin } from "@expo/config-plugins";
 import { withAppBuildGradle } from "@expo/config-plugins";
 
-const withAndroidSourcemaps: ConfigPlugin<void> = (config) => {
-  return withAppBuildGradle(config, async (config) => {
-    const appBuildGradle = config.modResults;
-    if (appBuildGradle.contents.match("datadog-sourcemaps.gradle")) {
+import { SourceMapUploadOptions } from "../getErrorTrackingPluginsFromOptions";
+
+const withAndroidSourcemaps =
+  (options: SourceMapUploadOptions): ConfigPlugin<void> =>
+  (config) => {
+    return withAppBuildGradle(config, async (config) => {
+      const appBuildGradle = config.modResults;
+      if (appBuildGradle.contents.match("datadog-sourcemaps.gradle")) {
+        return config;
+      }
+
+      appBuildGradle.contents = appBuildGradle.contents.replace(
+        /apply plugin\: \"com\.facebook\.react\"/,
+        `apply plugin: "com.facebook.react"\napply from: "${require("path").dirname(
+          require.resolve("@datadog/mobile-react-native/package.json")
+        )}/datadog-sourcemaps.gradle"`
+      );
+
+      if (!options.serviceName) {
+        return config;
+      }
+
+      // preserves the config set by withAndroidProguardMappingFiles
+      if (appBuildGradle.contents.match("datadog {")) {
+        appBuildGradle.contents = appBuildGradle.contents.replace(
+          /    checkProjectDependencies = "none"/,
+          [
+            `    checkProjectDependencies = "none"`,
+            `    serviceName = "${options.serviceName}"`,
+          ].join("\n")
+        );
+        return config;
+      }
```

Comment:
As you said in your description, this can be brittle and break if we change the order of the plugins or change options in the proguard mapping files plugin.
It will also make changes to this part of the code harder, for instance if we ever want to add more options in the future.
Finally, in the case where we only want to upload proguard mapping files and not android sourcemaps, we'll be missing the `serviceName` option.

I would suggest to create a dedicated `withAndroidConfiguration` for adding the configuration snippet:

```groovy
datadog {
    checkProjectDependencies = "none"
    serviceName = "com.myapp"
}
```

It should go first in the order of plugins with a comment to explicitely put it before any other android plugin.

Reply:
Great suggestion @louiszawadzki.

I updated the PR to move this configuration to its own plugin.

A couple of callouts.

1. I am not entirely sure if adding the plugin inside of `getErrorTrackingPluginsFromOptions` is the best place.  An alternative is I could move it higher up to the main `plugin/index.ts` if that is preferred. 
2. With this change we will now always set the config for `checkProjectDependencies = "none"` while previously it was only set if you had `androidProguardMappingFiles` enabled